test/deprecated/io/checkopenmem.chpl

```diff
@@ -0,0 +1,94 @@
+use IO except openmem; // to test deprecation warning for openmem
```

Comment:
Shouldn't this test check the error messages for not putting `use IO` at all?

It seems unrealistic to have `use IO except openmem`.

Ditto for other tests in this directory.

Reply:
The `open*` functions' arguments require types that trigger compilation errors instead of warnings, so I was trying to get their specific warning message.  Maybe that means I shouldn't bother adding compilation warnings for those functions at all, though, since in the basic case the user will need a use to call them in any real sense . . .

Reply:
Right, I'd be in favor of trimming down the set of things we're adding warnings for (so that it covers almost all practical use cases)

Reply:
Okay, I trimmed three tests and dropped the warning message for open.  I chose not to update the openmem test for two reasons:
1. That was going to be a lot of work and the point of me copying the test in the first place was to easily verify that the wrapped function behaved as expected rather than having a hidden error that a simple compilation would not detect
2. I had seen some problems where symbols that should have been allowed in when the use was limited in some way would not behave appropriately, so having a test when a limited use of IO was present would hopefully allow us to catch things a user would run into if they were already trying to limit what was brought it but had missed a symbol because we were including it by default.

Obviously reason 1 is mostly laziness, but I think reason 2 is sufficient to justify being lazy in this particular case, considering I updated the other ones.  Thoughts?

Reply:
OK.